docs/Users/Tutorials/Linter_Bears_Advanced.rst

```diff
@@ -0,0 +1,134 @@
+Linter Bears - Advanced Feature Reference
+=========================================
+
+Often linters are no easy tools. To squeeze out the last bit of functionality
+and efficiency, ``@linter`` provides some advanced features ready for use.
+
+Supplying Configuration Files with ``generate_config``
+------------------------------------------------------
+
+Sometimes tools require a configuration file to run. ``@linter`` supports that
+easily by overriding ``generate_config()``.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file):
+            config_file = ("value1 = 1\n"
+                           "value=2 = 2")
+            return config_file
+
+The string returned from this method is written into a temporary file before
+invoking ``create_arguments()``. If you return ``None``, no configuration file
+is generated.
```

Comment:
please mention that you can make this a non static method if you whish to use bear logging